Update hardhat guide (#32)
lukso-network/lukso-playground: smart-contracts-hardhat/README.md

# Basic Sample Hardhat Project

This project is used as a guide to show how to setup HardHat to work with the **[LUKSO networks](https://docs.lukso.tech/networks/mainnet/parameters)** and **[LSPs](https://docs.lukso.tech/contracts/introduction)**.

[HardHat](https://hardhat.org/docs) is a development environment to:

- **create** and **edit**
- **compile** and **debug**
- **deploy** and **verify**

smart contracts on EVM based blockchains.

> Further information can be found on our [Hardhat Guide](https://docs.lukso.tech/learn/smart-contract-developers/getting-started).

## Setup

Install the dependencies

```bash

> npm i

Reviewer: npm or bun ?

I see a bun lock file

Is this folder using bun or npm?

If bun let's use bun cli for the scripts

```

Set the private environment variables

```bash
cp .env.example .env
```

> > **INFO** Make sure to add the private key of an EOA and Universal Profile address for deployment.

Reviewer: I would reco to remove the "Universal Profile address" part.

A new user will not have this info avail. yet and this might block them in going further. Moreover, this is not needed for the first tutorials. I suggest to add this "Universal Profile address" where the tutorial requires it and not as a "global dependency" let's say

## Development

### Compile Contracts

Enforce compilation of contracts, and display the detailed stack traces for debugging.

```bash
npm run build
```

Regular compilation of your smart contracts.

```bash
npx hardhat compile
```

> **INFO** Add the `--verbose` and `--show-stack-traces` flags for further information

### Deploy Scripts

Deploy a contract as an Externally Owned Account (EOA)

```bash
npx hardhat --network luksoTestnet run scripts/deployEOA.ts
```

Deploy a contract as a Universal Profile (UP)

```bash
npx hardhat --network luksoTestnet run scripts/deployUP.ts
```

> **INFO** Adjust the network and token name within the command and script.

### Verify Contracts

```bash
npx hardhat verify myTokenAddress --constructor-args ./verify/myTokenName.ts --network luksoTestnet
```

## Getting Started

> The syntax of smart contracts works by **inheritance**, similarly as Object Oriented Programming language. Every object represents an individual smart contracts, which inherits underlying functionalities. If you create a new smart contract using LSPs, you can `import` standardized and modular presets from the `@lukso/lsp-smart-contracts` developer library. Such presets can then be combined to create complex contract deployments like Universal Profiles and Digital Assets.

Reviewer: > The syntax of smart contracts works by **inheritance**, similarly as Object Oriented Programming language. Every object represents an individual smart contracts, which inherits underlying functionalities.

this is not needed

> # Basic Sample Hardhat Project

Reviewer 2: Suggest giving it a more specific title maybe? This is what I had in my repo + it's the default when you bootstrap a Hardhat project, but it sounds very generic for the LUKS Playground.

@Hugoo @fhildeb ideas?

> This project is used as a guide to show how to setup HardHat to work with the **[LUKSO networks](https://docs.lukso.tech/networks/mainnet/parameters)** and **[LSPs](https://docs.lukso.tech/contracts/introduction)**.

Reviewer 2: ```suggestion
This project is used as a guide to show how to setup HardHat to interact with **[LSPs](https://docs.lukso.tech/contracts/introduction)** smart contracts on **[LUKSO networks](https://docs.lukso.tech/networks/mainnet/parameters)**.
```

> [HardHat](https://hardhat.org/docs) is a development environment to:

Reviewer 2: I would delete completely this section + bullet points. This LUKSO Playground is intended to focus on LUKSO tools usage, not to explain what Hardhat is.
People coming into this repo should already know what Hardhat is and how to use it. If they don't they could go the Hardhat docs to learn these.
